### ISSUE TITLE
Visual Studio deadlocked

### ISSUE THREAD

**Issue by tomgilder** [CodeConnect/AliveFeedback#90]
Visual Studio just froze on me, and it looks like Alive has caused it.

VS's main thread stack trace looks like this:

```
WindowsBase.dll!System.Windows.Threading.DispatcherSynchronizationContext.Wait(System.IntPtr[] waitHandles, bool waitAll, int millisecondsTimeout)   
                        mscorlib.dll!System.Threading.SynchronizationContext.InvokeWaitMethodHelper(System.Threading.SynchronizationContext syncContext, System.IntPtr[] waitHandles, bool waitAll, int millisecondsTimeout)     
                        [Native to Managed Transition]   
                        [Managed to Native Transition]   
                        mscorlib.dll!System.Threading.Monitor.Wait(object obj, int millisecondsTimeout, bool exitContext)    
                        mscorlib.dll!System.Threading.Monitor.Wait(object obj, int millisecondsTimeout)  
                        mscorlib.dll!System.Threading.ManualResetEventSlim.Wait(int millisecondsTimeout, System.Threading.CancellationToken cancellationToken)   
                        mscorlib.dll!System.Threading.Tasks.Task.SpinThenBlockingWait(int millisecondsTimeout, System.Threading.CancellationToken cancellationToken)     
                        mscorlib.dll!System.Threading.Tasks.Task.InternalWait(int millisecondsTimeout, System.Threading.CancellationToken cancellationToken)     
                        mscorlib.dll!System.Threading.Tasks.Task<bool>.GetResultCore(bool waitCompletionNotification)    
                        mscorlib.dll!System.Threading.Tasks.Task<bool>.Result.get()  
                        CodeConnect.ExecutionEngine.dll!CodeConnect.ExecutionEngine.MethodGraph.MethodInfo.MethodInfo(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.CSharp.Syntax.BaseMethodDeclarationSyntax baseMethodSyntax, Microsoft.CodeAnalysis.IMethodSymbol methodSymbol)    
                        CodeConnect.ExecutionEngine.dll!CodeConnect.ExecutionEngine.MethodGraph.MethodInfo.GetMethodInfoAtPositionAsync(string filePath, Microsoft.VisualStudio.Text.Span span)  
                        mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.__Canon>.Start<CodeConnect.ExecutionEngine.MethodGraph.MethodInfo.<GetMethodInfoAtPositionAsync>d__40>(ref CodeConnect.ExecutionEngine.MethodGraph.MethodInfo.<GetMethodInfoAtPositionAsync>d__40 stateMachine)   
                        CodeConnect.ExecutionEngine.dll!CodeConnect.ExecutionEngine.MethodGraph.MethodInfo.GetMethodInfoAtPositionAsync(string filePath, Microsoft.VisualStudio.Text.Span span)  
                        ProjectOrange.dll!CodeConnect.ProjectOrange.Views.AliveAdornment.TextBufferChangedEventHandler.AnonymousMethod__50_0()   
                        mscorlib.dll!System.Lazy<System.Threading.Tasks.Task<CodeConnect.ExecutionEngine.MethodGraph.MethodInfo>>.CreateValue()  
                        mscorlib.dll!System.Lazy<System.Threading.Tasks.Task<CodeConnect.ExecutionEngine.MethodGraph.MethodInfo>>.LazyInitValue()    
                        mscorlib.dll!System.Lazy<System.Threading.Tasks.Task<CodeConnect.ExecutionEngine.MethodGraph.MethodInfo>>.Value.get()    
                        ProjectOrange.dll!CodeConnect.ProjectOrange.Views.AliveAdornment.setLocalStateAndUpdateIcon()    
                        mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<CodeConnect.ProjectOrange.Views.AliveAdornment.<setLocalStateAndUpdateIcon>d__47>(ref CodeConnect.ProjectOrange.Views.AliveAdornment.<setLocalStateAndUpdateIcon>d__47 stateMachine)   
                        ProjectOrange.dll!CodeConnect.ProjectOrange.Views.AliveAdornment.setLocalStateAndUpdateIcon()    
                        ProjectOrange.dll!CodeConnect.ProjectOrange.Views.AliveAdornment.Pipeline_ExecutionDataUpdated(object sender, System.EventArgs e)    
                        mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<CodeConnect.ProjectOrange.Views.AliveAdornment.<Pipeline_ExecutionDataUpdated>d__49>(ref CodeConnect.ProjectOrange.Views.AliveAdornment.<Pipeline_ExecutionDataUpdated>d__49 stateMachine)     
                        ProjectOrange.dll!CodeConnect.ProjectOrange.Views.AliveAdornment.Pipeline_ExecutionDataUpdated(object sender, System.EventArgs e)    
                        CodeConnect.Analytics.dll!CodeConnect.Analytics.EventWrapper.SimpleAsyncWrappedHandler.handler(object s, System.EventArgs e)     
                        mscorlib.dll!System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start<CodeConnect.Analytics.EventWrapper.SimpleAsyncWrappedHandler.<handler>d__2>(ref CodeConnect.Analytics.EventWrapper.SimpleAsyncWrappedHandler.<handler>d__2 stateMachine)   
                        CodeConnect.Analytics.dll!CodeConnect.Analytics.EventWrapper.SimpleAsyncWrappedHandler.handler(object s, System.EventArgs e)     
                        ProjectOrange.dll!CodeConnect.ProjectOrange.Pipeline.RecentExecutionData.set(CodeConnect.ExecutionEngine.ExecutionData value)    
                        ProjectOrange.dll!CodeConnect.ProjectOrange.Pipeline.Process()   
                        mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(object stateMachine)   
                        mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)    
                        mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)    
                        mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()     
                        mscorlib.dll!System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation..cctor.AnonymousMethod__8_0(object state)    
                        WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)   
                        WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)  
                        WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeImpl()    
                        WindowsBase.dll!System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(object state)   
                        mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)    
                        mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)    
                        mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state)  
                        WindowsBase.dll!System.Windows.Threading.DispatcherOperation.Invoke()    
                        WindowsBase.dll!System.Windows.Threading.Dispatcher.ProcessQueue()   
                        WindowsBase.dll!System.Windows.Threading.Dispatcher.WndProcHook(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)   
                        WindowsBase.dll!MS.Win32.HwndWrapper.WndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam, ref bool handled)  
                        WindowsBase.dll!MS.Win32.HwndSubclass.DispatcherCallbackOperation(object o)  
                        WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)   
                        WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler)  
                        WindowsBase.dll!System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority priority, System.TimeSpan timeout, System.Delegate method, object args, int numArgs)    
                        WindowsBase.dll!MS.Win32.HwndSubclass.SubclassWndProc(System.IntPtr hwnd, int msg, System.IntPtr wParam, System.IntPtr lParam)   
```

**Comment by AmadeusW** [comment]
That's definitely Alive. I'm very sorry about this. I'll investigate what happened.
Is there anything specific that you were doing when this happened? Is there anything special about the method or overall code that you were at?

**Comment by tomgilder** [comment]
Was just typing, not entirely sure what bit of code I was in at the time.

Love the speedy reply :smile:

**Comment by AmadeusW** [comment]
Okay. I'll see what I can do! I can't promise that the fix will be as speedy ;)

**Comment by AmadeusW** [comment]
I have a fix, but without a repro we can just hope that it is the right fix. We will do more testing now.

**Comment by JoshVarty** [comment]
Thanks for your help, this was fixed in the most recent release 1.3.8.
